### DISCUSSION TITLE
Thoughts on the new Pydantic AI?

### DISCUSSION THREAD

**Discussion by Eigenwise** [567-labs/instructor#1235] [General]
https://ai.pydantic.dev/

Seems like what Pydantic is working on is very similar to Instructor.

What are your thoughts, are there any plans of further collaboration or will Instructor somehow start integrating PydanticAI or will it stay completely separate or..?

**Comment by timothytrust** [comment]
I watched some intro video on PydanticAI & what I really liked is that it not only validates the LLMs output but also lets you validate the inputs via pydantic model.

Of course you could do this by yourself, but Pydantic AI offers this out of the box and it really lets you control your workflow even more -> input is valid, output is valid, everything in between is only LLM.

Pretty neat stuff.

**Reply by Eigenwise** [reply]
I mean, that's not really anything you'd have to do yourself, right? You can't create an invalid instance of a Pydantic object, it would throw an error, no?

**Reply by timothytrust** [reply]
Yes right - Sure, you can of course do this yourself. Its just a nice design idea, which I was not super aware of yet.

**Reply by Eigenwise** [reply]
No what I mean is that, unless I am missing something, this is baked-in into pydantic itself, so you'd never have to do it yourself regardless of using pydanticai or something that uses "normal" pydantic

**Comment by Eigenwise** [comment]
Either way I'd love some input from @jxnl on this!

**Comment by Eigenwise** [comment]
Bump for visibility... Was this ever discussed at all? @jxnl

**Comment by arreumb** [comment]
It's like 'pydantic is all you need' is becoming more and more true with each passing year. However at the moment  I much prefer the simplicity and flexibility of Instructor (I will also be eternally grateful to it for saving me from LangChain).
